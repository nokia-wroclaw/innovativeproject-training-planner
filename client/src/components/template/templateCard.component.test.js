import React from 'react';
import renderer from 'react-test-renderer';
import TemplateCard, {setCardColor} from './templateCard.component';
import {MemoryRouter} from 'react-router-dom';

const templatelistItemMock = {
  userName: ['mitrainingplaner@gmail.com'],
  _id: '5e8cf389fe13d0205e4ae533',
  date: 'Thu Apr 30 2020',
  startTime: '06:37 PM',
  endTime: '06:43 PM',
  trainingType: 'Soft Skills Training',
  instructor: 'Jan Nowak',
  title: 'Jak rozmawiać z ludźmi z grypy, by się nie pozabijać',
  description: 'Spotkanie dla ludzi mających trudności z komunikacją w grupie',
  willLearn: 'Jak pracować dobrze w grupie.',
  mustKnow: 'blank :)',
  materials: 'blank',
  createdAt: '2020-04-07T21:41:29.903Z',
  updatedAt: '2020-04-30T15:38:33.528Z',
  __v: 0,
};

test('Match TemplateCard to snapshot', () => {
  const outcome = renderer.create(
      <MemoryRouter>
        <TemplateCard item={templatelistItemMock} />
      </MemoryRouter>,
  ).toJSON();
  expect(outcome).toMatchSnapshot();
});

test('Generate TemplateCard background color', () => {
  const particularColor = setCardColor('Hardware Training');
  const anyColor = setCardColor('Big Papa Pillow');

  expect(particularColor).toBe('#7c4dff');
  expect(anyColor).toBe('#9e9e9e');
});
