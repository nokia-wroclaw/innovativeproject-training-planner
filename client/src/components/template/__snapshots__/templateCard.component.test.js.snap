// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match TemplateCard to snapshot 1`] = `
<div
  className="card hoverable soft-training"
>
  <div
    className="card-reveal"
  >
    <span
      className="card-title grey-text text-darken-4"
    >
      Jan Nowak
    </span>
    <p>
      Thu Apr 30 2020
    </p>
  </div>
  <div
    className="card-content white-text"
  >
    <div
      className="row"
    >
      <div
        className="col s10 offset-s1"
      >
        <span
          className="card-title activator truncate"
        >
          Jak rozmawiać z ludźmi z grypy, by się nie pozabijać
        </span>
        <p
          className="truncate"
        >
          Spotkanie dla ludzi mających trudności z komunikacją w grupie
        </p>
      </div>
      <div
        className="col s1"
      >
        <i
          className="material-icons right"
        >
          group
        </i>
      </div>
    </div>
    <br />
    <div
      className="divider"
    />
  </div>
  <div
    className="modal"
    id="5e8cf389fe13d0205e4ae533"
  >
    <div
      className="modal-footer"
    >
      <a
        className="modal-close btn-flat"
        href="#!"
      >
        <i
          className="material-icons gray"
        >
          close
        </i>
      </a>
    </div>
    <div
      className="modal-content "
    >
      <div
        className="center"
      >
        <div
          className="details-bar"
        >
          <a
            className="modal-close btn-flat"
            href="/"
            onClick={[Function]}
          >
            <i
              className="material-icons red-text text-lighten-1 left"
            >
              delete
            </i>
            DELETE
          </a>
          <a
            className="modal-close btn-flat "
            href="/inviteTemplate/5e8cf389fe13d0205e4ae533"
            onClick={[Function]}
          >
            <i
              className="material-icons left"
            >
              edit
            </i>
            EDIT
          </a>
          <a
            className="modal-close btn-flat"
            href="/sendInvite/5e8cf389fe13d0205e4ae533"
            onClick={[Function]}
          >
            <i
              className="material-icons left"
            >
              mail
            </i>
            SEND
          </a>
        </div>
      </div>
      <div
        className="divider blue"
      />
      <div
        className="details"
      >
        <html
          lang="en"
          xmlns="http://www.w3.org/1999/xhtml"
        >
          <head>
            <meta
              content="text/html; charset=utf-8"
              httpEquiv="Content-Type"
            />
            <meta
              content="width=device-width, initial-scale=1.0"
              name="viewport"
            />
            <title>
              Invitation
            </title>
          </head>
          <body
            style={
              Object {
                "MsTextSizeAdjust": "100%",
                "WebkitTextSizeAdjust": "100%",
                "margin": 0,
                "padding": 0,
                "width": "100%",
              }
            }
          >
            <table
              align="left"
              border="0"
              cellPadding={0}
              cellSpacing={0}
              height="100%"
              valign="top"
              width="100%"
            >
              <tbody>
                <tr>
                  <td
                    align="center"
                    valign="top"
                  >
                    <table
                      align="center"
                      border="0"
                      cellPadding={0}
                      cellSpacing={0}
                      valign="top"
                      width="600"
                    >
                      <tbody>
                        <table
                          align="left"
                          border="0"
                          cellPadding={0}
                          cellSpacing={0}
                          valign="top"
                          width="600px"
                        >
                          <tbody>
                            <tr>
                              <td
                                align="left"
                              >
                                <span
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 30,
                                      "fontWeight": undefined,
                                      "lineHeight": "30px",
                                    }
                                  }
                                >
                                  Jak rozmawiać z ludźmi z grypy, by się nie pozabijać
                                </span>
                              </td>
                            </tr>
                            <tr>
                              <td
                                align="left"
                              >
                                <span
                                  style={
                                    Object {
                                      "color": "gray",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 10,
                                      "fontWeight": "bold",
                                      "lineHeight": "10px",
                                    }
                                  }
                                >
                                  INSTRUCTOR: 
                                </span>
                                <br />
                                <span
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 15,
                                      "fontWeight": undefined,
                                      "lineHeight": "15px",
                                    }
                                  }
                                >
                                  Jan Nowak
                                </span>
                              </td>
                            </tr>
                            <tr>
                              <td
                                align="left"
                              >
                                <span
                                  style={
                                    Object {
                                      "color": "gray",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 10,
                                      "fontWeight": "bold",
                                      "lineHeight": "10px",
                                    }
                                  }
                                >
                                  DATE: 
                                </span>
                                <br />
                                <span
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 15,
                                      "fontWeight": undefined,
                                      "lineHeight": "15px",
                                    }
                                  }
                                >
                                  Thu Apr 30 2020
                                </span>
                                <br />
                                <span
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 15,
                                      "fontWeight": undefined,
                                      "lineHeight": "15px",
                                    }
                                  }
                                >
                                  06:37 PM - 06:43 PM
                                </span>
                              </td>
                            </tr>
                            <tr>
                              <td
                                align="left"
                              >
                                <span
                                  style={
                                    Object {
                                      "color": "gray",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 10,
                                      "fontWeight": "bold",
                                      "lineHeight": "10px",
                                    }
                                  }
                                >
                                  DESCRIPTION: 
                                </span>
                                <br />
                                <span
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 15,
                                      "fontWeight": undefined,
                                      "lineHeight": "15px",
                                    }
                                  }
                                >
                                  Spotkanie dla ludzi mających trudności z komunikacją w grupie
                                </span>
                              </td>
                            </tr>
                            <tr>
                              <td
                                align="left"
                              >
                                <span
                                  style={
                                    Object {
                                      "color": "gray",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 10,
                                      "fontWeight": "bold",
                                      "lineHeight": "10px",
                                    }
                                  }
                                >
                                  AGENDA: 
                                </span>
                                <br />
                                <span
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 15,
                                      "fontWeight": undefined,
                                      "lineHeight": "15px",
                                    }
                                  }
                                />
                              </td>
                            </tr>
                            <tr>
                              <td
                                align="left"
                              >
                                <span
                                  style={
                                    Object {
                                      "color": "gray",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 10,
                                      "fontWeight": "bold",
                                      "lineHeight": "10px",
                                    }
                                  }
                                >
                                  WHAT YOU WILL LEARN: 
                                </span>
                                <br />
                                <span
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 15,
                                      "fontWeight": undefined,
                                      "lineHeight": "15px",
                                    }
                                  }
                                >
                                  Jak pracować dobrze w grupie.
                                </span>
                              </td>
                            </tr>
                            <tr>
                              <td
                                align="left"
                              >
                                <span
                                  style={
                                    Object {
                                      "color": "gray",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 10,
                                      "fontWeight": "bold",
                                      "lineHeight": "10px",
                                    }
                                  }
                                >
                                  WHAT YOU MUST ALREADY KNOW: 
                                </span>
                                <br />
                                <span
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 15,
                                      "fontWeight": undefined,
                                      "lineHeight": "15px",
                                    }
                                  }
                                >
                                  blank :)
                                </span>
                              </td>
                            </tr>
                            <tr>
                              <td
                                align="left"
                              >
                                <span
                                  style={
                                    Object {
                                      "color": "gray",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 10,
                                      "fontWeight": "bold",
                                      "lineHeight": "10px",
                                    }
                                  }
                                >
                                  MATERIALS: 
                                </span>
                                <br />
                                <span
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "sans-serif",
                                      "fontSize": 15,
                                      "fontWeight": undefined,
                                      "lineHeight": "15px",
                                    }
                                  }
                                >
                                  blank
                                </span>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </tbody>
                    </table>
                  </td>
                </tr>
              </tbody>
            </table>
          </body>
        </html>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col s6"
    >
      <a
        className="modal-trigger btn-flat"
        data-target="5e8cf389fe13d0205e4ae533"
        href="#!"
      >
        <i
          className="material-icons left"
        >
          event_note
        </i>
        DETAILS
      </a>
    </div>
    <div
      className="col s6"
    >
      <a
        className="btn-flat"
        href="/sendInvite/5e8cf389fe13d0205e4ae533"
        onClick={[Function]}
      >
        <i
          className="material-icons left"
        >
          mail
        </i>
        SEND
      </a>
    </div>
  </div>
  <br />
</div>
`;
